hazelcast/src/main/java/com/hazelcast/map/impl/MapContainer.java

```diff
@@ -271,7 +271,8 @@ ConstructorFunction<RecordFactoryAttributes, RecordFactory> createRecordFactoryC
         };
     }
 
-    public void initWanReplication(NodeEngine nodeEngine) {
+    public void initWanReplication() {
+        NodeEngine nodeEngine = mapServiceContext.getNodeEngine();
         WanReplicationRef wanReplicationRef = mapConfig.getWanReplicationRef();
```

Comment:
can you point me where the `mapConfig#wanReplicationRef` is changed? If we are not changing mapConfig reference, we will still get old one.

Reply:
wanReplicationRef isn't changed. Whole point of this PR is to

> Existing map/cache referencing to a non-existent wan replication, don't
replicate even after that non-existent wan replication is added
dynamically. This PR aims to change that.

So if you have 0 wan configuration and then add a map with wanRef to `some-non-existing-wan-repl` and then lastly add a wan replication: `some-non-existing-wan-repl` it should work now. About dynamically changing wanRef, I asked @vbekiaris about it but we decided against it for now since it requres a new API.